cylc/flow/tests/test_suite_files.py

```diff
@@ -181,51 +181,61 @@ def get_register_test_cases():
     ]
 
 
-class TestSuiteSrvFilesManager(unittest.TestCase):
+class TestSuiteFiles(unittest.TestCase):
 
-    def setUp(self):
-        self.suite_srv_files_mgr = SuiteSrvFilesManager()
-
-    @mock.patch('cylc.flow.suite_srv_files_mgr.os')
-    def test_register(self, mocked_os):
-        """Test the SuiteSrvFilesManager register function."""
+    @mock.patch('os.unlink')
+    @mock.patch('os.makedirs')
+    @mock.patch('os.symlink')
+    @mock.patch('os.readlink')
+    @mock.patch('os.path.isfile')
+    @mock.patch('os.path.isabs')
+    @mock.patch('os.getcwd')
+    @mock.patch('os.path.abspath')
+    @mock.patch('cylc.flow.suite_files.get_suite_srv_dir')
+    def test_register(
+            self,
+            mocked_get_suite_srv_dir,
+            mocked_abspath,
+            mocked_getcwd,
+            mocked_isabs,
+            mocked_isfile,
+            mocked_readlink,
+            mocked_symlink,
+            mocked_makedirs,
+            mocked_unlink
+    ):
+        """Test the register function."""
         def mkdirs_standin(_, exist_ok=False):
             return True
 
-        # we do not need to mock these functions
-        mocked_os.path.basename = os.path.basename
```

Comment:
With this change, my IDE linter flagged the `import os` above :arrow_up: as not used any more

Reply:
I think this is an out-of-date diff?

Reply:
Oh, weird. I thought I delete/checked out the code again. But did that just before logging off and calling it a day, not the best time for reviews. Sorry

Reply:
Np!

Reply:
Morning! Just going through notifications, then double-checked this one.

Looked again on the GitHub UI using "View file" option, and then directly at your branch: https://github.com/oliver-sanders/cylc/blob/suite-srv-files-mgr/cylc/flow/tests/test_suite_files.py

Right at the top `import os`, but if you search for `os.` using `CTRL+F`, there is probably only mocked occurrences of `@mock.patch('os.unlink')`.

I think the import is not being used, but if we need it because of the mocked imports, we would need a comment there to prevent (probably my future self with bad memory after running a linter) others of trying to remove it :+1:

Reply:
I think we don't need it for the mocked imports, so it can be removed.  (I just did a test, to be sure).

Reply:
Mock performs imports backstage, learn something new every day.